### ISSUE TITLE
Low durability hammer + anvil crash

### ISSUE THREAD

**Issue by fuami** [anegostudios/VintageStory-Issues#1263]
Version: v1.15.3 and v1.15.4-rc.1 
Platform: Windows
Modded: Yes / but reproduced after disabling everything.

### Description
Crash while attempting to look at an anvil with a mostly destroyed hammer.

### How to reproduce
1. Pull out my 66 / 1800 Durability Bismuth bronze hammer.
2. Look at my bismuth bronze anvil ( crashes )

Switching to a different hammer doesn't crash, its just this one.

### Expected behavior
Not to crash. :)

### Logs
```
Running on 64 bit Windows with 16 GB RAM 
Version: v1.15.4-rc.1 (Unstable)8/8/2021 2:59:04 PM: Critical error occurred
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Vintagestory.Client.NoObf.HudHotbar.UpdateCurrentToolMode(Int32 newSlotIndex)
   at Vintagestory.Common.EventManager.TriggerGameTick(Int64 ellapsedMilliseconds, IWorldAccessor world)
   at Vintagestory.Client.NoObf.ClientMain.MainRenderLoop(Single dt)
   at Vintagestory.Client.NoObf.ClientMain.MainGameLoop(Single deltaTime)
   at _iTgescuUf0kXKNXu6dfxHeIMvfG._a7jaYnEQ6vTTh1meiPfEAUCGJVlA(Single )
   at _igaury40RDGQNtdY51dKy1LFQaG._PsCGkT7lMIncUD4FDdqauQ3AaTl(Single )
   at _igaury40RDGQNtdY51dKy1LFQaG._1lUbSLybYeKj0UTZLOdfdtQwqRJ(Single )
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
   at OpenTK.GameWindow.DispatchRenderFrame()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _Yit8BJyfwe0pl1vnZXdnsUXBHeK._qejIe9laMG7hmFafani6nftUNCU(_OtB8vvTzxo67cRDgokUIe37bIfO , String[] )
   at _lO2cybHaCglaEphC0FED3uFtHRh._qejIe9laMG7hmFafani6nftUNCU(ThreadStart )
-------------------------------

Event Log entries containing Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 8/1/2021 6:01:53 PM, Site = , Source = Windows Error Reporting, Message = Fault bucket 1501095903599259191, type 4
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.15.3.0
P3: 61051c0c
P4: KERNELBASE.dll
P5: 10.0.19041.1110
P6: 4809adf2
P7: c000041d
P8: 0000000000034ed9
P9: 
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER9CAA.tmp.dmp
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER9FD8.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER9FE9.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER9FE7.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERA007.tmp.txt

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_Vintagestory.exe_d8cbbebc34e1fad331edc1a8cd9893c9f9bf40_c6d645e5_e0aac734-2026-4fd0-8292-132d93762d13

Analysis symbol: 
Rechecking for solution: 0
Report Id: 5b317817-c72a-4b94-8329-f97a4a6e9cb0
Report Status: 268435456
Hashed bucket: 97ac2779c8f4055624d4f6c56bd3ee37
Cab Guid: 0 }
--------------
{ TimeGenerated = 8/1/2021 6:01:51 PM, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.15.3.0, time stamp: 0x61051c0c
Faulting module name: KERNELBASE.dll, version: 10.0.19041.1110, time stamp: 0x4809adf2
Exception code: 0xc000041d
Fault offset: 0x0000000000034ed9
Faulting process id: 0x710c
Faulting application start time: 0x01d78728f2adc7bd
Faulting application path: C:\Users\Fuami\AppData\Roaming\Vintagestory\Vintagestory.exe
Faulting module path: C:\WINDOWS\System32\KERNELBASE.dll
Report Id: 5b317817-c72a-4b94-8329-f97a4a6e9cb0
Faulting package full name: 
Faulting package-relative application ID:  }
--------------
{ TimeGenerated = 7/24/2021 2:36:18 PM, Site = , Source = Windows Error Reporting, Message = Fault bucket 1859817290734047358, type 5
Event Name: RADAR_PRE_LEAK_64
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.15.2.0
P3: 10.0.19042.2.0.0
P4: 
P5: 
P6: 
P7: 
P8: 
P9: 
P10: 

Attached files:
\\?\C:\Users\Fuami\AppData\Local\Temp\RDR8797.tmp\empty.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER87B7.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER87C8.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER87E5.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER8815.tmp.txt

These files may be available here:


Analysis symbol: 
Rechecking for solution: 0
Report Id: 81366b7d-9696-4692-a835-96f4fd7292d3
Report Status: 268435456
Hashed bucket: 009d070bf886c81e29cf65f8e155107e
Cab Guid: 0 }
```

**Comment by Ragolution** [comment]
I can't seem to easily reproduce this. Is there anything else you could've done beforehand that triggered the crash?

**Comment by tyronx** [comment]
any chance you have or had a mod active that modifies hammers?

**Comment by copygirl** [comment]
Yeah I was thinking it might be something similar, though I did not think about mods. I suppose it could have been a mod that adds additional smithing modes or so? It would still be worth adding a check and reverting to the first available mode when this occurs.

**Comment by tyronx** [comment]
yea, indeed

**Comment by fuami** [comment]
As I mentioned above, the crash occurs without any mods installed.

That said however, Its possible that I had a mod at some point that added a mode, I tested a mod the other day briefly and have since removed it. So its possible that the hammer somehow got set to a mode that no long exists.

**Comment by fuami** [comment]
I tried diving into this further to see if I could get a better reproduction process with the mod that caused it, however my own research into the problem has been fruitless. None of the mods that I have, or have tested seem to add an additional hammer mode.

Even after re-adding all of the mods, the hammer still causes the same crash.

Its possible that one of the mods I've used is still responsible, but as best I can determine its at least not quite as simple as an additional mode that has been removed.

I've only really raised questions, rather then answering them, but it seems like the plan was to make the logic behind modes more robust regardless, so it probobly doesn't make much a difference.
